man/man1/mbnavadjust.1
.TH mbnavadjust 1 "24 June 2010" "MB-System 5.0" "MB-System 5.0"
.SH NAME
\fBmbnavadjust\fP - Package that solves for optimal navigation by
matching bathymetry of overlapping swaths.

.SH VERSION
Version 5.0

.SH SYNOPSIS
\fBmbnavadjust\fP [\fB-V -H\fP] 

.SH DESCRIPTION
\fBMBnavadjust\fP is an interactive graphical program used to
adjust swath data navigation by matching bathymetric features
in overlapping and crossing swaths. The primary purpose of
\fBmbnavadjust\fP is to eliminate relative navigational errors in swath
data obtained from poorly navigated sonars. Submerged platforms
such as towed vehicles, remotely operated vehicles (ROVs) and
autonomous underwater vehicles (AUVs) are frequently not navigated 
with accuracy equivalent to the lateral resolution of 
swath bathymetry obtained with high frequency sonars operated
close to the seafloor. This is  particularly true for systems
that depend on ultra-short baseline navigation (USBL) or on
inertial navigation (as opposed to long baseline (LBL) navigation
obtained using transponder networks). Many old, pre-GPS surveys with 
hull mounted sonars also suffer from poor navigation relative to the 
resolution of the swath bathymetry, and can thus be improved with 
application of this tool. \fBMBnavadjust\fP also works well to 
co-register surveys at different scales. For instance, our
initial use of this tool involved co-registering a deep-towed Reson
8101 multibeam survey of the Loihi Seamount summit (offshore Hawaii)
with a hull mounted Simrad EM300 multibeam survey. 

When swath surveys are poorly navigated, features clearly imaged
by the sonar may not match in overlapping and crossing swaths.
However, by systematically determining the positional offsets between
matching features throughout a survey, it is possible to solve,
or invert, for a navigation model which is consistent with the
original navigational constraints, satisfies the requirement
of reasonableness, and also causes features in overlapping swaths
to match within the sonar's resolution. The analysis and adjustment
can be limited to lateral (x and y or longitude and latitude) offsets
when all vertical offsets have been corrected prior to importation
into \fBMBnavadjust\fP (e.g. tidal corrections have been made). 
Otherwise, the offset analysis and inversions are three dimensional.

Previous attempts to address this problem (e.g. Nishimura et al. [1988])
have focused on automating the identification of matching features
and their navigational offsets. The automated approach is 
problematic because seafloor features are complex, and the
cross correlation function of matching features often exhibits
multiple local minima. Our approach is to automate the identification
of matching features and the ultimate solution for an optimal
indentification while depending on interactive determination of
the navigational offsets. We have not found an automatic algorithm
that can reliably replace the human eye and brain in correctly 
lining up real seafloor features.

\fBMBnavadjust\fP operates within the context of the \fBmbprocess\fP
parallel processing scheme, but it should used on data for which
the initial navigation processing (e.g. \fBmbnavedit\fP is complete.
When swath data files are imported into \fBmbnavadjust\fP, the data is
taken from processed files generated by \fBmbprocess\fP whenever
possible (when those files exist). If no processed file exists,
the raw data is used. When the processing with \fBmbnavadjust\fP
is completed and an optimal navigation solution is achieved, new
navigation files are generated (".na#" files, see below for full
description) which supercede the ".nve" files
generated by \fBmbnavedit\fP. Rerunning \fBmbprocess\fP will merge
the new navigation into a new set of processed files. Once
a swath file is processed with \fBmbnavadjust\fP, any further attempts
to process the navigation with \fBmbnavedit\fP will only produce
navigation files that are ignored by \fBmbprocess\fP. This behavior
can be reversed manually using \fBmbset\fP, but then the user will
unnecessarily complicating his or her efforts.

The inversion for an optimal navigation which fits the offsets identified
at matching features must also be reasonable. This is achieved by
minimizing perturbations to speed and acceleration in the inversion.
This approach of penalizing first and second derivatives within
a linear inverse problem is common within the geophysical
inverse theory literature (e.g. Parker [1994]).

Users are advised that \fBmbnavadjust\fP is complicated in both
conception and implementation. We strongly recommend that users
read all of the documentation provided below prior to attempting
a first use of this software.
The meaning and use of the individual widgets and windows employed in
the graphical interface is provided in the \fBINTERACTIVE CONTROLS\fP
section. The \fBUSING MBNAVADJUST\fP section provides a more coherent 
discussion of how to use \fBmbnavadjust\fP, how this program interacts
with the \fBmbprocess\fP parallel processing scheme, and the underlying
concepts and algorithms.

.SH AUTHORSHIP
David W. Caress (caress@mbari.org)
.br
  Monterey Bay Aquarium Research Institute
.br
Dale N. Chayes (dale@ldeo.columbia.edu)
.br
  Lamont-Doherty Earth Observatory

.SH OPTIONS
.TP
.B \-H
This "help" flag cause the program to print out a description
of its operation and then exit immediately.
.TP
.B \-V
Normally, \fBmbnavadjust\fP outputs nothing to the stderr stream.
If the \fB-V\fP flag is given, then \fBmbnavadjust\fP 
works in a "verbose" mode and
outputs the program version being used, all error status messages, 
and a large amount of other information.

.SH INTERACTIVE CONTROLS

.SS MBnavadjust Window
This is the primary window of \fBmbnavadjust\fP.

.TP
.B File
This button activates a pulldown menu with six items:
\fBNew\fP, \fBOpen\fP, \fBClose\fP, 
\fBImport Swath Data\fP, and \fBQuit\fP. 

.TP
.B File->New
This menu item brings up a popup window which allows the user to
specify a new \fBmbnavadjust\fP project. As part of initializing
a new project, \fBmbnavadjust\fP creates a file with a ".nvh"
suffix and a directory with a ".dir" suffix.
.TP
.B File->Open
This menu item brings up a popup window which allows the user to
open an existing \fBmbnavadjust\fP project.
.TP
.B File->Close
This menu item causes \fBmbnavadjust\fP to close the current project.
.TP
.B File->Import Swath Data
This menu item brings up a popup window which allows the user to
specify a swath sonar data file or a list of 
swath files to be imported into the current
\fBmbnavadjust\fP project.
This program uses the \fBMBIO\fP library and 
will read any swath sonar 
formats supported by \fBMBIO\fP. A list of the swath sonar data
formats currently supported by \fBMBIO\fP and their
identifier values is given in the \fBMBIO\fP manual page. 
If the swath sonar data file is named using the \fBMB-System\fP
suffix convention (format 11 files end with ".mb11", format
41 files end with ".mb41", etc.), then the program will
automatically use the appropriate format id; otherwise the
format must be set by the user in the format text box.
.TP
.B File->Quit
This button causes \fBmbnavadjust\fP to quit. If a project has been
created or opened, that project will be closed before the program
quits.
.TP
.B Option
This button brings up a pulldown menu with one item: \fBControls\fP.
.TP
.B Option->Controls
This menu item brings up a dialog window with several slider widgets
that set parameters controlling how swath data are imported and displayed.
These widgets are discussed in the \fBControls\fP section below.
 .TP
.B View
This button brings up a pulldown menu with fifteen items in three 
groups. This menu controls the contents of the list widget in the lower
right \fBMBnavadjust Window\fP. If an \fBmbnavadjust\fP project
has been created or opened, the \fBData Table\fP list 
widget will display a table of items controlled by these menu items, 
and the corresponding menu item will be disabled. Selection of another item
items will change what is displayed in the \fBData Table\fP list
widget accordingly.

The first group includes the \fBShow Surveys\fP, 
\fBShow Data Files\fP, \fBShow Data File Sections\fP, \fBShow All Crossings\fP, 
\fBShow All Crossings\fP, \fBShow >25% Crossings\fP, \fBShow >50% Crossings\fP,
\fBShow True Crossings\fP, and \fBShow Ties\fP items.
.TP
.B View->Show Surveys
This menu item causes the \fBData Table\fP list widget to display a list of the 
surveys in the current \fBmbnavadjust\fP project. Within \fBMBnavadjust\fP, a 
survey is a group of swath files that are contiguous in time.
.TP
.B View->Show Data Files
This menu item causes the \fBData Table\fP list widget to display a list of the swath
data files in the current \fBmbnavadjust\fP project. 
.TP
.B View->Show Data File Sections
This menu item causes the \fBData Table\fP list widget to display a list of the swath
data sections in the current \fBmbnavadjust\fP project. 
.TP
.B View->Show All Crossings
This menu item causes the \fBData Table\fP list widget to display a list of all the swath
crossings in the current \fBmbnavadjust\fP project. 
When this option is set, selecting a list item causes \fBmbnavadjust\fP to load and display the selected
selected crossing in the "Nav Err" window.
.TP
.B View->Show >25% Crossings
This menu item causes the \fBData Table\fP list widget to display a list of all the swath
crossings in the current \fBmbnavadjust\fP project for which the two sections overlap by 
more than 25%. 
When this option is set, selecting a list item causes \fBmbnavadjust\fP to load and display the selected
selected crossing in the "Nav Err" window.
.TP
.B View->Show >50% Crossings
This menu item causes the \fBData Table\fP list widget to display a list of all the swath
crossings in the current \fBmbnavadjust\fP project for which the two sections overlap by 
more than 50%.. 
When this option is set, selecting a list item causes \fBmbnavadjust\fP to load and display the selected
selected crossing in the "Nav Err" window.
.TP
.B View->Show True Crossings
This menu item causes the \fBData Table\fP list widget to display a list of all the swath
crossings  in the current \fBmbnavadjust\fP project for which the navigation tracks cross. 
When this option is set, selecting a list item causes \fBmbnavadjust\fP to load and display the selected
selected crossing in the "Nav Err" window.
.TP
.B View->Show Ties
This menu item causes the \fBData Table\fP list widget to display a list of the tie points
in the current \fBmbnavadjust\fP project. 

The second group includes the 
\fBShow All Surveys\fP, \fBShow Only Selected Survey\fP, \fBShow Only Selected File\fP, 
and \fBShow Only Selected Section\fP items. One of these options is always active, and
modifies what is displayed in the \fBData Table\fP list.
.TP
.B View->Show All Surveys
This menu item allows all of the possible items (file, sections, crossings, ties)
to be displayed in the \fBData Table\fP list. 
.TP
.B View->Show Only Selected Surveys
This menu item limits the items (file, sections, crossings, ties) displayed in 
the \fBData Table\fP list to those associated with the most recently
selected survey (e.g. a survey selected by clicking in the \fBData Table\fP list 
while the <View->Show Surveys> option is active).
.TP
.B View->Show Only Selected File
This menu item limits the items (file, sections, crossings, ties) displayed in 
the \fBData Table\fP list to those associated with the most recently
selected file (e.g. a file selected by clicking in the \fBData Table\fP list 
while the <View->Show Files> option is active).
.TP
.B View->Show Only Selected Section
This menu item limits the items (file, sections, crossings, ties) displayed in 
the \fBData Table\fP list to those associated with the most recently
selected section (e.g. a section selected by clicking in the \fBData Table\fP list 
while the <View->Show Sections> option is active).

The third group includes two options:
.TP
.B View->Visualize Survey
This option is not active, and is for future use.
.TP
.B View->Show Modelplot
This option causes an addition "Model Plot" window to be shown that displays
the navigation adjustment model as three time series: longitude on top, latitude
in the middle, and vertical on the bottom.
.TP
.B Action
This button brings up a pulldown menu with nine items: 
\fBSet File Poor Navigation\fP, \fBSet File Good Navigation\fP, 
\fBSet File Fixed Navigation\fP, \fBAuto-Pick Offsets\fP, 
\fBCheck for New Crossings\fP, \fBAnalyze Crossings\fP, \fBZero All Z Offsets\fP, 
\fBInvert Navigation\fP, and \fBApply Adjusted Navigation\fP.
.TP
.B Action->Set File Poor Navigation
This menu item causes \fBmbnavadjust\fP to treat the navigation of
a selected swath data file as poor. This option is only activated
when the \fBData Table\fP list widget shows a list of the imported
swath files and one file has been selected by clicking in that list.
In the case where no files are fixed, the final adjusted navigation model
is centered on the average navigation offsets required for good files.
The offsets tieing to files with poor navigation are ignored in 
calculating the averages.
.TP
.B Action->Set File Good Navigation
This menu item causes \fBmbnavadjust\fP to treat the navigation of
a selected swath data file as good, which is in fact the default state
for swath files. This option is only activated
when the \fBData Table\fP list widget shows a list of the imported
swath files and one file has been selected by clicking in that list.
In the case where no files are fixed, the final adjusted navigation model
is centered on the average navigation offsets required for good files.
The offsets tieing to files with good navigation are used in 
calculating the averages.
.TP
.B Action->Set File Fixed Navigation
This menu item causes \fBmbnavadjust\fP to treat the navigation of
a selected swath data file as fixed, which means that all navigation
points for this file will have zero offsets in the final adjusted navigation. 
Effectively, all of the offsets between fixed and unfixed files are taken up in
the adjustments of the unfixed files.
This option is only activated
when the \fBData Table\fP list widget shows a list of the imported
swath files and one file has been selected by clicking in that list.
.TP
.B Action->Auto-Pick Offsets
This menu item causes \fBmbnavadjust\fP to automatically pick offsets in 
all of the unset crossings currently displayed in the \fBData Table\fP list.
If <View->Show True Crossings> has been selected, then only true crossings are
displayed and only the unset true crossings will be automatically picked.
In general, unsupervised offset picks are frequently erroneous, and use
of this experimental option is not encouraged.
.TP
.B Action->Check For New Crossings
This menu item causes \fBmbnavadjust\fP to search for new crossings using
the adjusted navigation from the current inversion. Often, new places where
swaths overlap and cross emerge as the navigation is adjusted. The percentage
of overlap is also recalculated for all of the existing crossings. If new true
crossings are found, the user will be unable to perform a new inversion until
all of the true crossings have been analyzed.
.TP
.B Action->Analyze Crossings
This menu item causes \fBmbnavadjust\fP to load and display the first
crossing in the "Nav Err" window.
.TP
.B Action->Zero All Z Offsets
This menu item causes \fBmbnavadjust\fP to load and display the first
crossing in the "Nav Err" window.
.TP
.B Action->Invert Navigation
This menu item causes \fBmbnavadjust\fP to invert for an optimal
navigation solution that fits the offsets defined at the navigation
tie points while minimizing speed and acceleration. This option is
only activated when all of the true crossings in the current
\fBmbnavadjust\fP project have been analyzed interactively.
.TP
.B Action->Apply Adjusted Navigation
This menu item causes \fBmbnavadjust\fP to export the adjusted
navigation model from the current inversion to the
swath files in the project.This option is
only activated when the inversion is current, meaning that no new
data have been imported and no changes to offset picks have been 
made since the last inversion.
.TP
.B About
This button brings up a pulldown menu with one item: \fBAbout\fP.
.TP
.B About->About
This menu item brings up a popup window that displays the program's
name, authors, and version. Clicking the \fBDismiss\fP button will
close the window.
.TP
.B Data Table
This list widget displays one of five tables of information regarding
the current \fBmbnavadjust\fP project. These tables are a list of the
imported swath surveys, the data files, the shorter data sections 
comprising each data file, a list of the identified crossing swaths, and
a list of the navigation tie points defined interactively by the user.
The \fBView\fP pulldown menu allows the user to set which table is displayed.

The survey list will look something like:
.br
 	00 53 2009/08/03 08:18:49.484999 2009/08/03 22:52:59.375000
 	01 51 2009/08/04 09:03:11.938999 2009/08/04 23:02:03.470999
 	02 01 1998/05/13 01:33:36.791000 1998/05/13 02:42:11.703999
.br
Here the first column is the survey counter, the second is the number
of swath files included in each survey, and the following information
consists of the start and end times of the data in each survey
shown in YYYY/MM/DD HH:MM:SS.SSSSSS format.

The file list will look something like:
.br
 	0000:00 gd   11  0.0  0.0 ../../20090803/20090803_081706.mb88
 	0001:00 gd   11  0.0  0.0 ../../20090803/20090803_083332.mb88
 	0002:00 gd   11  0.0  0.0 ../../20090803/20090803_085004.mb88
 	0003:00 gd   11  0.0  0.0 ../../20090803/20090803_090636.mb88
 	0004:00 gd   10  0.0  0.0 ../../20090803/20090803_092307.mb88
 	 	..........................................
 	0053:01 gd   12  0.0  0.0 ../../20090804/20090804_090127.mb88
 	0054:01 gd   10  0.0  0.0 ../../20090804/20090804_092036.mb88
 	0055:01 gd   11  0.0  0.0 ../../20090804/20090804_093707.mb88
 	0056:01 gd   10  0.0  0.0 ../../20090804/20090804_095339.mb88
 	0057:01 gd   11  0.0  0.0 ../../20090804/20090804_101010.mb88
 	 	..........................................
 	0104:02 fx    8  0.0  0.0 ../../MBARI/1998em300/mbari_1998_630_msn.mb57
.br
Here the first column is the file counter and survey counter separated by a colon.
The second column indicates the file navigations state; "gd" indicates good
navigation, "pr" indicates poor navigation, and "fx" indicates fixed
navigation. The third column shows the number of sections extracted from this file. 
The fourth and fifth columns show any heading
or roll bias offsets in degrees applied to the swath data for that file. 
The sixth column gives the name of the swath data file imported into
\fBmbnavadjust\fP. Note that the name shown here is that of the "raw"
swath file. The data imported by \fBmbnavadjust\fP is, 
if possible, extracted from a "processed" swath file 
generated by \fBmbprocess\fP rather than the
associated "raw" file.

The file section list will look something like:
.br
 	00:0000:00 2009/08/03 08:17:07.546998 2009/08/03 08:18:49.484999
 	00:0000:01 2009/08/03 08:18:49.984999 2009/08/03 08:20:26.952999
 	00:0000:02 2009/08/03 08:20:27.452999 2009/08/03 08:22:05.890999
 	00:0000:03 2009/08/03 08:22:06.390999 2009/08/03 08:23:43.344001
 	00:0000:04 2009/08/03 08:23:43.844001 2009/08/03 08:25:19.796999
 	00:0000:05 2009/08/03 08:25:20.296999 2009/08/03 08:26:57.265997
 	00:0000:06 2009/08/03 08:26:57.765997 2009/08/03 08:28:35.219001
 	00:0000:07 2009/08/03 08:28:35.719001 2009/08/03 08:30:16.155999
 	00:0000:08 2009/08/03 08:30:16.655999 2009/08/03 08:31:57.594001
 	00:0000:09 2009/08/03 08:31:58.094001 2009/08/03 08:33:36.546999
 	00:0000:10 2009/08/03 08:33:37.046999 2009/08/03 08:33:37.546999
.br
Here the first column shows the section id with the survey counter, the file counter,
and the section counter separated by colons. The following information
consists of the start and end times of each section
shown in YYYY/MM/DD HH:MM:SS.SSSSSS format.

The crossing list will look something like:
 	-X    0 000:009 001:000  21  0
 	-     1 001:009 002:000  10  0
 	-     2 002:009 003:000  10  0
 	U     3 003:009 004:000   6  0
 	-     4 005:009 006:000  13  0
 	U     5 007:009 008:000   3  0
 	U     6 009:008 009:010   2  0
 	*     7 009:008 010:000  16  1
 	*X    8 009:009 010:000  41  1
 	-     9 009:008 010:001   1  0
.br
Here the first column indicates the processing status for the
crossing. The first character is the status flag. If the status flag is "U", 
then no decision has been made about skipping or tieing this crossing. 
New crossings always show a "U" flag prior to being
inspected by a user. If the first character is "-", then the crossing
has been skipped, and if the first character is "*", then at least one
tie point has been set. The second column is the crossing counter. The
third and fourth columns identify the swath data sections that overlap
in this crossing. Each are identified by their file id and section id
separated by ":". The fifth column indicates the percentage of overlap
of the two sections in this crossing. The larger the degree of overlap,
the more likely that diagnostic matching topographic features exist that
can be used to determine the navigation offsets required for this crossing.
The sixth, and last column gives the number of tie
points that have been defined for each crossing. New crossings always
begin with 0 tie points prior to being inspected by a user.

The tie point list will look something like:
.br
 	   7 0 009:008:07 010:000:04 00:00  1.02 -2.87 0.00 |  9.61  8.49 1.40 | 0.041 0.107 0.027
 	   8 0 009:009:02 010:000:00 00:00  0.90 -4.47 0.00 |  7.37  6.30 2.29 | 0.040 0.184 0.033
 	  12 0 009:008:04 010:002:04 00:00  4.11 -8.24 0.00 |  9.28  5.99 1.80 | 0.037 0.021 0.025
 	  14 0 010:000:02 010:002:06 00:00  2.12 -4.38 0.00 |  7.50  6.70 1.10 | 0.007 0.065 0.005
 	  16 0 009:007:04 010:003:04 00:00  5.90 -5.90 0.00 |  8.66  6.84 1.70 | 0.019 0.085 0.027
 	  19 0 009:006:04 010:004:04 00:00  8.32 -6.83 0.00 |  8.78  8.55 6.65 | 0.026 0.066 0.024
 	  21 0 009:005:05 010:005:03 00:00  8.96 -8.75 0.00 | 12.38  8.74 5.16 | 0.008 0.045 0.008
 	  23 0 009:004:04 010:006:04 00:00 12.23 -5.98 0.00 |  7.03  4.49 1.40 | 0.019 0.034 0.010
 	  26 0 009:003:04 010:007:04 00:00 17.42 -4.36 0.00 | 13.92 10.11 1.80 | 0.025 0.037 0.028
.br
Here the first column indicates the crossing which contains the tie point,
and the second column shows which tie point (of those defined for that
crossing) is displayed in a particular line. The third and fourth columns
identify the navigation control points of the tie point. The navigation
control points are specified by file, section, and nav point numbers separated
by ":". The fifth through seventh columns are the longitude, latitude and vertical
offsets (in meters) set interactively by the user. These represent the distance the second
navigation control point must be moved relative to the first in order to make
the bathymetry in the two swaths match. The tenth through twelth columns show the 
magnitude of the three axes of the uncertainly ellipsoid associated with each tie.
The uncertainly ellipsoid is estimated as a 3x3 tensor and used to weight the tie
offsets in the navigation adjustment inversion. Here the major and second axes are
always close to horizontal, and the minor is axis is always close to vertical.
The last three columns are nonzero only
after an inversion for an optimal navigation solution has been performed. These
represent the residual, or difference, between the offset calculated for this tie
point in the inversion and that set by the user 
(displayed in the fifth through seventh columns).

If the file list is displayed and one file is selected by clicking in
the list, then the user can fix or unfix the navigation of that file using
the <Action->Fix File> or <Action->Unfix File> pulldown menu items.
If either the crossing list or the tie point list are displayed, selecting
one crossing or one tie point by clicking in the list widget will cause
the specified crossing to be loaded and displayed in the \fBNav Err\fP
window.

.SS Controls Window
This dialog window is brought up by clicking the \fBOption->Controls\fP
menu item in the \fBMBnavadjust Window\fP. This window includes two slider
widgets setting data importation parameters, four slider widgets setting 
data display parameters, a button to apply any changes, and a button to
dismiss the window.

.TP
.B Max Section Length (km)
When \fBmbnavadjust\fP imports swath data, it breaks each data file
up into a set of sections, or pieces. The geographical coverages of all sections 
are then compared to determine where swaths overlap or cross. This control
sets the section track length. The ideal section length varies with the sonar
altitude and ping rate. If the section length is too short, there will be too
many small sections and an excessive number of crossings to analyze. If the
section length is too large, individual sections will be slow to load and display.
.TP
.B Max # Soundings in Section
This slider widget sets an additional or alternative control on swath data section size: the
maximum number of soundings (not pings) that may be included in a given section. For some
datasets, it may be more convenient to use the number of soundings rather than the track
length as the criterea for breaking files up into sections.
.TP
.B Decimation
When swath bathymetry sections are displayed in the \fBNav Err\fP window,
the primary display consists of bathymetric contours. 
The time required for calculating these
contours is linear with the number of soundings. This slider allows users to 
decimate the pings used for the contour displays. If the decimation is set
to 1, all soundings are used. If the decimation is set to 2, the soundings from
every second ping are used for contouring.
.TP
.B Contour Interval (m)
This slider widget sets the depth interval in meters for the bathymetric contours.
.TP
.B Color Interval (m)
This slider widget sets the the depth interval in meters at which the
color of the bathymetric contours changes. The contours are generated in four
colors: black, red, green, and blue. The color interval should be a multiple
of the contour interval.
.TP
.B Tick Interval (m)
This slider widget sets the depth interval in meters at which contours have
downhill facing tickmarks. The tick interval should be a multiple of the contour
interval.
.TP
.B Inversion Smoothing
This slider widget sets the importance of smoothing in the adjusted
navigation model. Larger values yield a smoother model, smaller values a rougher model.
The default is 4.00, and possible ranges are 0.10 to
10.0, where the smoothing penalty weight applied to the first and
second derivatives of the adjusted navigation perturbation is ten
raised to this power.
.TP
.B Apply
This button applies any changed values in the slider widgets of the \fBControls Window\fP.
If the \fBControls Window\fP is closed without the \fBApply\fP button being clicked, all
changes to the values controlled by the sliders are lost.
.TP
.B Dismiss
This button closes the \fBControls Window\fP.

.SS Nav Err Window
This window allows users to interactively inspect crossing swaths,
define navigation tie points where bathymetric features can be matched,
and set the navigation offsets at those tie points. This window can
be brought up in a number of ways. If one pulls down the
menu option <Action->Analyze Crossings>, then the "Nav Error" window 
will come up with the first crossing loaded. Alternatively, if the
<View->Show All Crossings> menu item has been selected so that a table of
crossings is displayed, clicking once on any of the crossing items in
this tabel will bring up the  "Nav Error" window with that crossing loaded.
Similarly, if tie points have been established and the tie point table
displayed by selecting the <View->Show Ties> menu item, then clicking on
any of the tie items this table will bring up the "Nav Error" window 
and load the crossing that includes the selected tie point. If the
"Nav Error" window is already displayed, clicking on crossing or tie items
in the display tables will load the selected crossing in place of whatever
crossing was previously shown.

The Nav Err window includes a number of button widgets and three display
canvases. The larger display to the right shows bathymetric contour maps
of the overlapping swaths overlain by navigation tracks and any tie points
that have been defined for the current crossing. The smaller canvas on the
middle left shows the RMS bathymetry misfit between the two swaths as a
function of lateral (x and y) offset using the current vertical (z)offset. The
smallest canvas on the lower left shows the RMS Bathymetry misfit between
the two swaths as a function of vertical offset using the current
lateral offset.

.TP
.B Contour Display
This canvas displays color bathymetric contours from both overlapping
swaths in the current swath crossing. Initially, the display is scaled
so that the entire area covered by the current crossing is visible. The 
bathymetric contours of the second swath can be moved relative to those
of the first by dragging with the left mouse button in the canvas. The
current navigational offset changes as the contours shift. The location 
of the current lateral offset also changes simultaneously on the misfit plot. The
user can display a subregion of the current map by dragging a rectangle in the canvas 
with the middle mouse button. This zoom operation can be repeated as many times
as desired. The right mouse button is used to set the location of the current
navigational tie point. When the right mouse button is clicked on a point in
the contour map, the closest navigation control points from each swath to 
that position are selected for the current tie point. In practice, the user
should right-click on the bathymetric feature being matched to set the
tie point correctly.
.TP
.B Lateral Misfit Display
This canvas shows a color two dimensional plot of the RMS bathymetric misfit
between the two overlapping swaths. The misfit is shown as a function
of relative lateral (x and y) offsets between the two  swaths using the
current vertical (z) offset.  Put
another  way,  the misfit plot shows how good, or bad, the
misfit becomes as one moves one swath relative to another.
The  lowest misfit values are shown in red; higher misfits
are shown in blues to purples. The location of the minimum
misfit  is  marked by a black "X", and the location of the
current navigation offset is shown by a small  red  square
with a black outline. If an inversion has been performed, a
small black "+" will mark the location of the offset calculated
by the inversion. If a user left-clicks in the Lateral Misfit Display, the
location of the cursor is taken as the new navigation offset
and the red square moves to that location. Simultaneously, the
bathymetric contours in the Contour Display shift to reflect
the new offset.
.TP
.B Vertical Misfit Display
This canvas shows a color profile plot of the RMS bathymetric misfit
between the two overlapping swaths. The misfit is shown as a function
of relative vertical (z) offset between the two  swaths using the
current lateral (x andy) offsets.  Put
another  way,  the misfit plot shows how good, or bad, the

trunk/src/mbnavadjust/mbnavadjust.c

/* Begin user code block <abstract> */
/* End user code block <abstract> */

/**
 * README: Portions of this file are merged at file generation
 * time. Edits can be made *only* in between specified code blocks, look
 * for keywords <Begin user code> and <End user code>.
 */
/*
 * Generated by the ICS Builder Xcessory (BX).
 *
 * BuilderXcessory Version 6.1.3
 * Code Generator Xcessory 6.1.3 (08/19/04) CGX Scripts 6.1 Motif 2.1 
 *
 */


/* Begin user code block <file_comments> */
#include <stdio.h>
#include <stdlib.h>
/* End user code block <file_comments> */

/*
 * Motif required Headers
 */
#include <X11/StringDefs.h>
#include <Xm/Xm.h>
#include <Xm/DialogS.h>
#include <Xm/RepType.h>
#include <Xm/MwmUtil.h>

/**
 * Globally included information.
 */


/*
 * Headers for classes used in this program
 */

/**
 * Common constant and pixmap declarations.
 */
#include "mbnavadjust_creation.h"

/**
 * Convenience functions from utilities file.
 */
extern void RegisterBxConverters(XtAppContext);
extern XtPointer BX_CONVERT(Widget, char *, char *, int, Boolean *);
extern XtPointer BX_DOUBLE(double);
extern XtPointer BX_SINGLE(float);
extern void BX_MENU_POST(Widget, XtPointer, XEvent *, Boolean *);
extern Pixmap XPM_PIXMAP(Widget, char **);
extern void BX_SET_BACKGROUND_COLOR(Widget, ArgList, Cardinal *, Pixel);

/**
 * Declarations for shell callbacks.
 */
extern void BxExitCB(Widget, XtPointer, XtPointer);

/* Begin user code block <globals> */

#include "mb_define.h"
#include "mb_status.h"
#include "mb_aux.h"
#include "mbnavadjust.h"

Widget CreatemainWindow(Widget parent);

/* End user code block <globals> */

/**
 * Change this line via the Output Application Names Dialog.
 */
#define BX_APP_CLASS "mbnavadjust"

int main( int argc, char **argv)
{
    Widget       parent;
    XtAppContext app;
    Arg          args[256];
    Cardinal     ac;
    Widget   topLevelShell;
    Widget   mainWindow;
    
    /* Begin user code block <declarations> */
    
    /* make sure that the argc that goes to XtVaAppInitialize
       is 1 so that no options are removed by its option parsing */
    int		 argc_save;
    argc_save = argc;
    argc = 1;
    
    /* End user code block <declarations> */
    
    /*
     * Initialize Xt. 
     */
    
    XtSetLanguageProc(NULL, (XtLanguageProc) NULL, NULL); 
    
    /*
     * The applicationShell is created as an unrealized
     * parent for multiple topLevelShells.  The topLevelShells
     * are created as popup children of the applicationShell.
     * This is a recommendation of Paul Asente & Ralph Swick in
     * _X_Window_System_Toolkit_ p. 677.
     */
    
    parent = XtVaOpenApplication ( &app, 
                                   BX_APP_CLASS, 
                                   NULL, 
                                   0, 
                                   &argc, 
                                   argv, 
                                   NULL, 
                                   sessionShellWidgetClass, 
                                   NULL );
    
    RegisterBxConverters(app);
    XmRepTypeInstallTearOffModelConverter();
    
    /* Begin user code block <create_shells> */
    /* End user code block <create_shells> */
    
    /*
     * Create classes and widgets used in this program. 
     */
    
    /* Begin user code block <create_topLevelShell> */
    /* End user code block <create_topLevelShell> */
    
    ac = 0;
    XtSetArg(args[ac], XmNtitle, "MBnavadjust"); ac++;
    XtSetArg(args[ac], XmNiconName, "MBnavadjust"); ac++;
    XtSetArg(args[ac], XmNx, 82); ac++;
    XtSetArg(args[ac], XmNy, 242); ac++;
    XtSetArg(args[ac], XmNwidth, 962); ac++;
    XtSetArg(args[ac], XmNheight, 400); ac++;
    topLevelShell = XtCreatePopupShell((char *)"topLevelShell",
        topLevelShellWidgetClass,
        parent,
        args, 
        ac);
    XtAddCallback(topLevelShell, XmNdestroyCallback, BxExitCB, (XtPointer)0);
    mainWindow = (Widget)CreatemainWindow(topLevelShell);
    XtManageChild(mainWindow);
    XtPopup(XtParent(mainWindow), XtGrabNone);
    
    /* Begin user code block <app_procedures> */

    /* initialize app value and wait until view realized */
    do_wait_until_viewed(app);

    /* initialize everything */
    do_mbnavadjust_init(argc_save,argv);

    /* End user code block <app_procedures> */
    
    /* Begin user code block <main_loop> */
    /* End user code block <main_loop> */
    
    XtAppMainLoop(app);
    
    /*
     * A return value regardless of whether or not the main loop ends. 
     */
     return(0); 
}
